File: node/samples/sample.ts
import { parseEmail } from '../dist/index.js';
import * as fs from 'fs';
import * as path from 'path';
import { fileURLToPath } from 'url';

const __dirname = path.dirname(fileURLToPath(import.meta.url));

// Resolve path to the shared root samples folder
const emailPath = path.resolve(__dirname, '../../samples/gmail.eml');
const email = fs.readFileSync(emailPath, 'utf-8');
const buffer = Buffer.from(email, 'utf-8');

const parsedEmail = await parseEmail(buffer);
console.log(parsedEmail);

File: python/samples/sample.py
import os
from lite_email_parser import parse_email

# Resolve path to the shared root samples folder
current_dir = os.path.dirname(os.path.abspath(__file__))
email_path = os.path.join(current_dir, '../../samples/sample1.eml')

with open(email_path, 'r', encoding='utf-8') as f:
    email_content = f.read()

result = parse_email(email_content)
print(result)
